draft-ietf-quic-tls.md

```diff
@@ -1025,6 +1025,35 @@ However, any tampering with the parameters will be detected
 when the handshake completes.
 
 
+## QUIC Max Crypto Data Extension {#max_crypto_data}
+
+When QUIC provides TLS messages via the CRYPTO_HS frame, one TLS message may be
+fragmented across different packets.  Thus TLS might have to buffer data.  TLS
+implementations may choose to limit the data they buffer before the handshake is
+completed and close the connection on receiving too much data.
+
+To communicate the maximum amount of data that TLS will allow to be sent in the
+CRYPTO_HS frames, TLS may use the `max_crypto_data` extension.
+
+~~~
+   enum {
+      max_crypto_data(TBD), (65535)
+   } ExtensionType;
+
+   struct {
+	   uint32 max_crypto_data;
+   } MaxCryptoData;
+~~~
+
+The `extension_data` field of the extension contains the MaxCryptoData
+structure.
+
+Implementations SHOULD send this extension.  Receivers do not need to process
+this extension.  The purpose of this extnesion is to provide a facility to debug
+issues during the handshake and also allow future extensibility of the protocol
```

Comment:
I would remove the note about debugging. It's primarily about controlling the amount of buffering and for future extensibility, right?

Reply:
its mostly about debugging